src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -949,6 +949,23 @@ def _wrapper(func):
         return _wrapper
 
 
+class ControllerAuthMixin(object):
+    @staticmethod
+    def _delete_token_cookie(token):
+        cherrypy.response.cookie['token'] = token
+        cherrypy.response.cookie['token']['expires'] = 0
+        cherrypy.response.cookie['token']['max-age'] = 0
+
+    @staticmethod
+    def _set_token_cookie(url_prefix, token):
+        cherrypy.response.cookie['token'] = token
+        if url_prefix == 'https':
+            cherrypy.response.cookie['token']['secure'] = True
+        cherrypy.response.cookie['token']['HttpOnly'] = True
+        cherrypy.response.cookie['token']['path'] = '/'
+        cherrypy.response.cookie['token']['SameSite'] = 'Strict'
+
+
```

Comment:
This is just a formal concern: we have a `services/` class called `AuthManager` which is already invoked from the `Auth` controller. Why creating a new class/mixing for this, when we already have a placeholder to group all auth-related operations?

OOP gives you the chance to bundle stuff by domains, but you need to learn and be consistent with the existing design, or otherwise there's no much benefit with encapsulation if the methods are spread across multiple related classes.

Reply:
@epuertat The refactoring proposal can be done in master and then backported to pacific.